cobalt/media/progressive/progressive_demuxer.cc

```diff
@@ -441,15 +441,18 @@ void ProgressiveDemuxer::Download(scoped_refptr<DecoderBuffer> buffer) {
     return;
   }
 
-  // copy timestamp and duration values
-  buffer->set_timestamp(requested_au_->GetTimestamp());
-  buffer->set_duration(requested_au_->GetDuration());
-
   // enqueue buffer into appropriate stream
   if (requested_au_->GetType() == DemuxerStream::AUDIO) {
     audio_demuxer_stream_->EnqueueBuffer(buffer);
   } else if (requested_au_->GetType() == DemuxerStream::VIDEO) {
-    video_demuxer_stream_->EnqueueBuffer(buffer);
+    // Copy everything to a new DecoderBuffer to overwrite data size.
+    scoped_refptr<DecoderBuffer> output_buffer =
```

Comment:
Currently, AllocateBuffer() allocates a DecoderBuffer to serve as a temporary memory buffer used by AvcAccessUnit::Read().  Consider allocating a normal buffer (i.e. uint8_t*) and read into it instead, and create a DecoderBuffer here using ExternalMemory based on the buffer, so avoid the extra allocation.